File: lab0/src/Person.java
package lab0.src;

public class Person {
  private String firstName;
  private String lastName;
  private int age;
  private boolean isMinor;

  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }

  public void setLastName(String lastName) {
    this.lastName = lastName;
  }

  public void setAge(int age) {
    if (age < 0) {
      System.out.println("Age cannot be negative");
      return;
    }

    this.age = age;

    if (age <= 17) {
      this.isMinor = true;
    } else {
      this.isMinor = false;
    }
  }

  public void printAge() {
    System.out.println(firstName + " " + lastName + " is " + age + " years old");
  }

  public void printIsMinor() {
    if (isMinor) {
      System.out.println(firstName + " " + lastName + " is a minor");
    } else {
      System.out.println(firstName + " " + lastName + " is not a minor");
    }
  }
}

File: lab2/files/Person.py
class Person:
    def __init__(self, first_name, last_name, age):
        self.first_name = first_name
        self.last_name = last_name
        self.age = age
        self.is_minor = age <= 17

    def print_age(self):
        print(f"{self.first_name} {self.last_name} is {self.age} years old")

    def print_is_minor(self):
        minor_status = "a minor" if self.is_minor else "not a minor"
        print(f"{self.first_name} {self.last_name} is {minor_status}")
